docs/aws/audit/codebuildmonitoring/rules/codepipeline_region_fanout.mdx

```diff
@@ -20,10 +20,381 @@ Performance Efficiency, Operational Excellence, Reliability, Security
+
+Note: AWS CodeBuild does not inherently support fanning out across regions. You would need to manually set up your build projects in each region. Therefore, this misconfiguration check is more about ensuring that you have followed best practices in setting up your build projects.
+</Accordion>
+
+<Accordion title='Using CLI'>
+1. First, you need to install and configure AWS CLI on your local machine. You can do this by following the instructions provided by AWS. Make sure you have the necessary permissions to access the resources.
+
+2. Once the AWS CLI is set up, you can list all the CodeBuild projects in your account using the following command:
 
+   ```
+   aws codebuild list-projects --region your-region
+   ```
+   Replace 'your-region' with the region you want to check. This command will return a list of all CodeBuild projects in the specified region.
+
+3. For each project, you can describe the project to get more details about it. Use the following command:
+
+   ```
+   aws codebuild batch-get-projects --names project-name --region your-region
+   ```
+   Replace 'project-name' with the name of the project you want to check and 'your-region' with the region where the project is located. This command will return detailed information about the project, including its configuration.
+
+4. Check the 'artifacts' section in the output of the previous command. If the 'location' field contains an S3 bucket that is not in the same region as the CodeBuild project, it means that the Code Deployment Pipeline is not fanning out across regions. This is a misconfiguration that should be corrected.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. Import the necessary AWS SDK for Python (Boto3) modules in your Python script. You will need the 'codebuild' module to interact with AWS CodeBuild.
+
+```python
+import boto3
+```
+
+2. Create a session using your AWS credentials. Replace 'aws_access_key_id', 'aws_secret_access_key', and 'aws_session_token' with your actual AWS credentials.
+
+```python
+session = boto3.Session(
+    aws_access_key_id='YOUR_ACCESS_KEY',
+    aws_secret_access_key='YOUR_SECRET_KEY',
+    aws_session_token='SESSION_TOKEN',
+)
+```
+
+3. Create a CodeBuild client from the session.
+
+```python
+codebuild = session.client('codebuild')
+```
+
+4. Now, you can use the 'list_projects' method to get a list of all CodeBuild projects in your account. For each project, use the 'batch_get_projects' method to get detailed information about the project, including the 'serviceRole' and 'artifacts' properties. Check if the 'location' property of the 'artifacts' object is in the same region as your CodeBuild project. If it's not, then the Code Deployment Pipeline is not fanning out across regions.
+
+```python
+projects = codebuild.list_projects()['projects']
+for project in projects:
+    project_details = codebuild.batch_get_projects(names=[project])['projects'][0]
+    project_region = project_details['arn'].split(':')[3]
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>